### PR TITLE
Improve Blink mechanism

### PR DESCRIPTION
**Pull request by casperx**

### CODE REVIEW

File: src/cascadia/TerminalCore/Terminal.hpp [hunk 1]
```diff
@@ -96,6 +96,8 @@ class Microsoft::Terminal::Core::Terminal final :
     COORD GetCursorPosition() const noexcept override;
     bool IsCursorVisible() const noexcept override;
     bool IsCursorOn() const noexcept override;
+    void SetCursorVisible(const bool isVisible) noexcept;
+	void SetCursorOn(const bool isOn) noexcept;
```
Comment (adiviness) [inline src/cascadia/TerminalCore/Terminal.hpp]: nit: indentation is off.

File: src/cascadia/TerminalCore/terminalrenderdata.cpp [hunk 2]
```diff
@@ -70,6 +70,16 @@ bool Terminal::IsCursorOn() const noexcept
...
 }
 
+void Terminal::SetCursorVisible(const bool isVisible) noexcept
+{
+    _buffer->GetCursor().SetIsVisible(isVisible);
+}
+
+void Terminal::SetCursorOn(const bool isOn) noexcept
+{
+	_buffer->GetCursor().SetIsOn(isOn);
```
Comment (adiviness) [inline src/cascadia/TerminalCore/terminalrenderdata.cpp]: nit: indentation is off. Is it a tab? We prefer spaces.

File: src/cascadia/TerminalControl/TermControl.cpp [hunk 3]
```diff
@@ -853,6 +878,17 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
...
 
+    // Method Description:
+    // - Toggle the cursor on and off when called by the cursor blink timer.
+    // Arguments:
+    // - sender: not used
+    // - e: not used
+    void TermControl::_BlinkCursor(Windows::Foundation::IInspectable const& /* sender */,
+                                   Windows::Foundation::IInspectable const& /* e */)
+    {
+        if (!(_terminal->GetTextBuffer().GetCursor().IsBlinkingAllowed() || _terminal->IsCursorOn()) || _terminal->IsCursorVisible()) _terminal->SetCursorOn(!_terminal->IsCursorOn());
```
Comment (adiviness) [inline src/cascadia/TerminalControl/TermControl.cpp]: Line is too long. Also, we require `{}` for the code block following an if statement.

Comment (binarycrusader) [reply src/cascadia/TerminalControl/TermControl.cpp]: What is the preferred line length for the Terminal project?

Comment (adiviness) [reply src/cascadia/TerminalControl/TermControl.cpp]: I don't think we have an actual hard number. I personally try to break a line once it's 100 cells wide. I assume clang-format has an option to enforce this so we'll be codifying an actual number here soon.

**Comment by mdtauk** [comment]
Is the cursor blink rate respecting the user's settings?

image

**Comment by casperx** [comment]
No, It doesn't. Now it blinks every 500 ms.

**Comment by oising** [comment]
> No, It doesn't. Now it blinks every 500 ms.

https://docs.microsoft.com/en-us/windows/desktop/api/winuser/nf-winuser-getcaretblinktime

*cough*

**Comment by casperx** [comment]
Yes, I had read the original CursorBlinker and I know about that function already. But I am not the first one who add the cursor blinking mechanism because I didn't know much about UWP. I came here to improve #686 and It isn't merge yet. Their code is changing now. I'll wait for final change.

**Comment by DHowett-MSFT** [comment]
Sorry, just so I understand:
This is a pull request that's intended to change @dkter's code, which we haven't yet merged, from #686?

I believe GitHub has a feature where you can suggest code changes to other peoples' pull requests, so that you can both collaborate on the same pull request.

**Comment by casperx** [comment]
I'm new to git. I didn't know I can do that. 😞

**Pull request closed without merging**